client/src/pages/page.cpp

```diff
@@ -419,13 +413,24 @@ void Page::initialize()
 
 void Page::slotItemContextMenuRequested(const QPoint &pos)
 {
+    QMenu *menu = handleContextMenu(pos);
+    if (menu) {
+        menu->exec(mUi.treeView->mapToGlobal(pos));
+        menu->deleteLater();
+    }
+}
+
+QMenu *Page::handleContextMenu(const QPoint &pos)
```

Comment:
Could we call this createContextMenu? (inspired by QLineEdit API). "Handle" sounds to me like acting on the user's selected action, while this is before that, it's about actually creating the menu.

Reply:
done